hazelcast/src/main/java/com/hazelcast/sql/impl/expression/datetime/ExtractField.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.expression.datetime;
+
+import java.time.OffsetDateTime;
+import java.time.temporal.ChronoField;
+import java.time.temporal.IsoFields;
+import java.time.temporal.WeekFields;
+
+public enum ExtractField {
+    CENTURY {
+        @Override
+        public double extract(OffsetDateTime time) {
+            int year = time.getYear();
+            int century = (year / YEARS_IN_CENTURY) + FIRST_CENTURY;
```

Comment:
This is not correct, it will return 21st century for `2000-12-31`, but 21st century started on `2001-1-1`. It also returns 1 for `-1-1-1`, while it should return -1. 

I suspect similar problems with DECADE, we need to consult the SQL standard and maybe PG implementation.

Reply:
And in `MILLENNIUM` :)

Reply:
I've learned one more thing :) 

I changed `MILLENNIUM` and `CENTURY` calculations to fix the one-off and BC dates problems.

I checked `DECADE` in Postgres and it seems the way how I implemented is the same as Postgres version.
From the docs:
```
Decade:
The year field divided by 10
```